### PR TITLE
WIP - Plugin Interface basic implementation using JSON

### PR DESCRIPTION
**Pull request by pritipath**
Deleting this and creating a new one that merges onto Juan's branch.

### CODE REVIEW

File: src/dotnet/CarbonAware/ICarbonAware.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+using System.Collections;
+
+namespace CarbonAware;
+
+public interface ICarbonAware
```
Comment (gfmatthews) [inline src/dotnet/CarbonAware/ICarbonAware.cs:5]: A thought occurs - is there a more precise name we could use for this interface? Like ```ICarbonAwarePlugin``` or even more like ```ICarbonAwareDataProvider```. I'm sorry I didn't think about this before in the design review but seeing it in usage made me think of this. Curious to hear your thoughts here

File: src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAware.Plugins.NewBasicJsonPlugin.csproj [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <ItemGroup>
+    <ProjectReference Include="..\CarbonAware\CarbonAware.csproj" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (gfmatthews) [inline src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAware.Plugins.NewBasicJsonPlugin.csproj:8]: The rest of this project is all dotnet 6 yes?

File: src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAwareNewBasicJsonPlugin.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections;
+using CarbonAware.Model;
+using Microsoft.Extensions.Logging;
+
+namespace CarbonAware.Plugins.NewBasicJsonPlugin;
+
+public class CarbonAwareNewBasicJsonPlugin : ICarbonAware
```
Comment (gfmatthews) [inline src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAwareNewBasicJsonPlugin.cs:7]: Why "New" basic json plugin?

File: src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAwareNewBasicJsonPlugin.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            Logger.LogInformation($"Key: {k} - Value: {props[k]}");
+        }
+        return await Task.Run(() => FakeData().Where((x) => filter(new QueryObject { 
+            Location = x.Location,
+            Time = x.Time,
+            Rating = x.Rating
+        })));
+    }
+
+    private IEnumerable<EmissionsData> FakeData()
```
Comment (gfmatthews) [inline src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAwareNewBasicJsonPlugin.cs:35]: I debate if perhaps the FakeData should be in its own file for clarity and to separate the hardcoded magic strings to a single location where its apparent what they are

**Review by gfmatthews — COMMENTED** [review]
Great progress y'all! Its so cool to see this getting in the actual code from the design!

File: src/dotnet/CarbonAware/ICarbonAware.cs (hunk 1 above)
Comment (juzuluag) [reply src/dotnet/CarbonAware/ICarbonAware.cs:5]: @gfmatthews , this still in work in progress. We need to flush out more details here in terms of naming convention and other things. More to come :-)

File: src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAwareNewBasicJsonPlugin.cs (hunk 4 above)
Comment (juzuluag) [reply src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAwareNewBasicJsonPlugin.cs:35]: @gfmatthews , as mentioned above, some more work needs to be done here.

File: src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAwareNewBasicJsonPlugin.cs (hunk 3 above)
Comment (juzuluag) [reply src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAwareNewBasicJsonPlugin.cs:7]: Same comment as above.

File: src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAware.Plugins.NewBasicJsonPlugin.csproj (hunk 2 above)
Comment (juzuluag) [reply src/dotnet/CarbonAware.Plugins.NewBasicJsonPlugin/CarbonAware.Plugins.NewBasicJsonPlugin.csproj:8]: Yes

File: src/dotnet/CarbonAware/ICarbonAware.cs (hunk 1 above)
Comment (pritipath) [reply src/dotnet/CarbonAware/ICarbonAware.cs:5]: Sorry, this was not meant to be a public PR yet.. Still WIP as Juan said.

**Pull request closed without merging**